lib/logstash/filters/csv.rb

```diff
@@ -105,7 +105,7 @@ def register
     @convert_symbols = @convert.inject({}){|result, (k, v)| result[k] = v.to_sym; result}
 
     # make sure @target is in the format [field name] if defined, i.e. surrounded by brakets
-    @target = "[#{@target}]" if @target && @target !~ /^\[[^\[\]]+\]$/
```

Comment:
Prior to this PR, this code was checking for `[field]` but would fail to match `[field][blah]`.